plugins/destination/postgresql/client/migrate.go

```diff
@@ -48,165 +168,193 @@ func (c *Client) Migrate(ctx context.Context, tables schema.Tables) error {
+	}
+	return nil
+}
 
-			// if this column contains primary key we will need to recreate the primary key
-			if c.enabledPks() && col.CreationOptions.PrimaryKey {
-				reCreatePrimaryKeys = true
-			}
-			sql := "alter table " + tableName + " drop column " + columnName
-			// right now we will drop the column and re-create. in the future we will have an option to automigrate
-			if _, err := c.conn.Exec(ctx, sql); err != nil {
-				return fmt.Errorf("failed to drop column %s on table %s: %w", col.Name, table.Name, err)
-			}
-			sql = "alter table " + tableName + " add column " + columnName + " " + columnType
-			if _, err := c.conn.Exec(ctx, sql); err != nil {
-				return fmt.Errorf("failed to add column %s on table %s: %w", col.Name, table.Name, err)
-			}
-		}
+func (c *Client) alterPKConstraint(ctx context.Context, pgTable *schema.Table, table *schema.Table) error {
+	c.logger.Info().Str("table", table.Name).Msg("Recreating primary keys")
+	tableName := pgx.Identifier{table.Name}.Sanitize()
 
-		// column exists and type is the same but constraints might differ
-		if c.enabledPks() && pgPKs[col.Name] != col.CreationOptions.PrimaryKey {
-			c.logger.Info().Str("table", table.Name).Str("column", col.Name).Bool("pk", col.CreationOptions.PrimaryKey).Msg("Column exists with different primary keys")
-			reCreatePrimaryKeys = true
+	tx, err := c.conn.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction to recreate primary keys: %w", err)
+	}
+	dropConstraintSQL := "alter table " + tableName + " drop constraint if exists " + pgTable.PkConstraintName
+	if _, err := tx.Exec(ctx, dropConstraintSQL); err != nil {
+		if err := tx.Rollback(ctx); err != nil {
+			c.logger.Error().Err(err).Msg("failed to rollback transaction")
 		}
+		return fmt.Errorf("failed to drop primary key constraint on table %s: %w", table.Name, err)
 	}
 
-	if reCreatePrimaryKeys {
-		c.logger.Info().Str("table", table.Name).Msg("Recreating primary keys")
-		if err := c.setNotNullOnPks(ctx, table); err != nil {
-			return fmt.Errorf("failed to enforce not null on primary keys: %w", err)
+	sql := "alter table " + tableName + " add constraint primary key (" + strings.Join(table.PrimaryKeys(), ",") + ")"
+	if _, err := tx.Exec(ctx, sql); err != nil {
+		if err := tx.Rollback(ctx); err != nil {
+			c.logger.Error().Err(err).Msg("failed to rollback transaction")
 		}
+		return fmt.Errorf("failed to add primary key constraint on table %s: %w", table.Name, err)
+	}
+	if err := tx.Commit(ctx); err != nil {
+		return fmt.Errorf("failed to commit transaction to recreate primary keys: %w", err)
+	}
+	return nil
+}
 
-		tx, err := c.conn.Begin(ctx)
-		if err != nil {
-			return fmt.Errorf("failed to begin transaction to recreate primary keys: %w", err)
+func (c *Client) autoMigrateTable(ctx context.Context, pgTable *schema.Table, table *schema.Table) error {
+	changedColumns, changedPgColumns := table.GetChangedColumns(pgTable)
```

Comment:
OK so might not be blocking but I think we should print all errors before suggesting `--force`.
Otherwise we might drop columns without explicitly warning the user about it when multiple columns require dropping

Reply:
mm, I think let's do it in a follow-up as it can also be a pretty long error message which can be also confusing.